zigbeelumi/integrationpluginzigbeelumi.cpp

```diff
@@ -230,38 +229,69 @@ void IntegrationPluginZigbeeLumi::setupThing(ThingSetupInfo *info)
         }
     }
 
+    if (thing->thingClassId() == lumiMotionSensor2ThingClassId) {
+        QTimer *presenceTimer = new QTimer(thing);
+        connect(presenceTimer, &QTimer::timeout, thing, [thing](){
+            thing->setStateValue(lumiMotionSensor2IsPresentStateTypeId, false);
+        });
+
+        // This one only sends an illuminage changed when a motion is detected.
+        ZigbeeCluster *lumiCluster = endpoint->getInputCluster(static_cast<ZigbeeClusterLibrary::ClusterId>(LUMI_CLUSTER_ID));
+        if (lumiCluster) {
+            connect(lumiCluster, &ZigbeeCluster::attributeChanged, thing, [thing, presenceTimer](const ZigbeeClusterAttribute &attribute){
+                switch (attribute.id()) {
+                case 0x0112: {
+                    quint32 lightIntensity = attribute.dataType().toUInt32();
+                    qCDebug(dcZigbeeLumi()) << thing << "Light intensity" << lightIntensity;
+                    thing->setStateValue(lumiMotionSensor2LightIntensityStateTypeId, lightIntensity);
+                    thing->setStateValue(lumiMotionSensor2IsPresentStateTypeId, true);
+                    thing->setStateValue(lumiMotionSensor2LastSeenTimeStateTypeId, QDateTime::currentDateTime());
```

Comment:
is this converting automatically to unix timestamp?

Reply:
fixed